pygeoapi_plugins/provider/sensorthings_edr.py

```diff
@@ -0,0 +1,401 @@
+        """
+        times, values = self._expand_observations(datastream)
+        thing = datastream['Thing']
+        coords = thing['Locations'][0]['location']['coordinates']
+        length = len(values)
+
+        return {
+            'type': 'Coverage',
+            'id': str(thing['@iot.id']),
+            'domain': {
+                'type': 'Domain',
+                'domainType': 'PointSeries',
+                'axes': {
+                    'x': {'values': [coords[0]]},
+                    'y': {'values': [coords[1]]},
+                    't': {'values': times},
+                },
+                'referencing': [GEOGRAPHIC_CRS, TEMPORAL_RS],
+            },
+            'ranges': {
+                id: {
+                    'type': 'NdArray',
+                    'dataType': 'float',
+                    'axisNames': ['t'],
+                    'shape': [length],
+                    'values': values,
+                }
+            },
+        }, length
+
+    @staticmethod
+    def _generate_paramters(datastream, label):
+        """
+        Generate parameters for a given datastream.
+
+        :param datastream: The datastream data to generate parameters for.
+        :param label: The label for the parameter.
+
+        :returns: A dictionary containing the parameter definition.
+        """
+        return {
+            'type': 'Parameter',
+            'description': {'en': datastream['description']},
+            'observedProperty': {'id': label, 'label': {'en': label}},
+            'unit': {
+                'label': {'en': datastream['unitOfMeasurement']['name']},
+                'symbol': datastream['unitOfMeasurement']['symbol'],
+            },
+        }
+
+    @staticmethod
+    def _make_dtf(datetime_):
+        """
+        Create a Date-Time filter for querying.
+
+        :param datetime_: Temporal filter.
+
+        :returns: A string datetime filter for use in queries.
+        """
+        dtf_r = []
+        if '/' in datetime_:
+            time_start, time_end = datetime_.split('/')
+            if time_start != '..':
+                dtf_r.append(f'phenomenonTime ge {time_start}')
+
+            if time_end != '..':
+                dtf_r.append(f'phenomenonTime le {time_end}')
+
+        else:
+            dtf_r.append(f'phenomenonTime eq {datetime_}')
+
+        return ' and '.join(dtf_r)
+
+    def _make_coverage_collection(self, response):
+        """
+        Build a CoverageCollection from the SensorThings API response.
+
+        :param response: source response from SensorThings.
+
+        :returns: The updated CoverageCollection object.
+        """
+        cc = {
+            'type': 'CoverageCollection',
+            'domainType': 'PointSeries',
+            'parameters': {},
+            'coverages': [],
+        }
+
+        for feature in response['value']:
+            try:
+                _ = feature['Datastreams'][0]
+            except IndexError:
+                continue
```

Comment:
I think it is more idiomatic to check if the list doesn't exist  rather than indexing in and checking the error.